lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -120,7 +125,19 @@ public function quote($input, $type = \PDO::PARAM_STR)
      */
     public function lastInsertId($name = null)
     {
-        return parent::lastInsertId($name);
+        if (null !== $name) {
+            return parent::lastInsertId($name);
+        }
+
+        $lastInsertId = parent::lastInsertId($name);
+
+        // The last insert ID is reset to "0" in certain situations by some implementations,
+        // therefore we keep the previously set insert ID locally.
+        if ($lastInsertId) {
```

Comment:
@Ocramius this PR does not touch sequence ID retrieval (which requires passing `$name`). So this is all about regular "autoincrement" via identity columns. I am not aware of the possibility to have an identity column generate an empty string or `0` (they start at `1`, don't they)?